int32[] data
time[] t